hazelcast/src/main/java/com/hazelcast/nio/tcp/WriteHandler.java

```diff
@@ -93,15 +94,32 @@ private void createWriter(String protocol) {
                 outputBuffer.put(stringToBytes(Protocols.CLUSTER));
                 registerOp(SelectionKey.OP_WRITE);
             } else if (Protocols.CLIENT_BINARY.equals(protocol)) {
+                adjustBufferSizesForClientConnection();
```

Comment:
intstead of adjusting at here (after `outputBuffer` is allocated at constructor), what about initialize it here?
I know, generally, these properties will not be configured so `outputBuffer` will be used as it created at constructor.
But for cases like too many client like 1000 clients, it will create `1000 * 32k` temporary output buffers and then they will be replaced with the adjusted ones.